thirdeye/thirdeye-frontend/app/pods/components/modals/entity-mapping-modal/component.js

```diff
@@ -0,0 +1,450 @@
+    this.builUrnToId(entities);
+
+    if (!url) {
+      return;
+    }
+    const relatedEntities = await fetch(url).then(checkStatus);
+
+    // merges createBy Props
+    relatedEntities.map((item) => {
+      if (!item.urn) {
+        return;
+      }
+      const { createdBy } = _.find(entities, { toURN: item.urn });
+      item.createdBy = createdBy;
+      return item;
+    });
+
+    set(this, '_relatedEntities', relatedEntities);
+  },
+
+  /**
+   * Fetches new entities with caching and diff checking
+  */
+  didReceiveAttrs() {
+    const {
+      metric,
+      _cachedMetric,
+      showEntityMappingModal
+    } = getProperties(this, 'metric', '_cachedMetric', 'showEntityMappingModal');
+
+    if (showEntityMappingModal && metric && !_.isEqual(metric, _cachedMetric)) {
+      set(this, '_cachedMetric', metric);
+
+      this._fetchRelatedEntities();
+    }
+  },
+
+  /**
+   * Checks if the currently selected entity is already in the mapping
+   */
+  mappingExists: computed(
+    'selectedEntity.label',
+    'relatedEntities.@each.alias',
+    'metric',
+    function() {
+      const {
+        selectedEntity: entity,
+        relatedEntities,
+        metric
+      } = getProperties(this, 'selectedEntity', 'relatedEntities', 'metric');
+
+      return (metric.label === entity.alias) || relatedEntities.some((relatedEntity) => {
+        relatedEntity.label === entity.alias;
+      });
+    }
+  ),
+
+  /**
+   * Helper function that sets an error
+   * @param {String} error - the error message
+   */
+  setError(error) {
+    set(this, 'errorMessage', error);
+  },
+
+  /**
+   * Helper function that clears errors
+  */
+  clearError() {
+    set(this, 'errorMessage', '');
+  },
+
+  /**
+   * Entity Mapping columns to be passed into
+   * ember-models-table
+   */
+  entityColumns: [
+    {
+      propertyName: 'type',
+      title: 'Types',
+      filterWithSelect: true,
+      className: 'te-modal__table-cell te-modal__table-cell--capitalized',
+      predefinedFilterOptions: [
+        'Metric',
+        'Other'
+      ]
+    },
+    {
+      template: 'custom/filterLabel',
+      title: 'Filter value',
+      className: 'te-modal__table-cell',
+      propertyName: 'label',
+      sortedBy: 'label',
+      // custom filter function
+      filterFunction(cell, string, record) {
+        return ['urn', 'label'].some(key  => record[key].includes(string));
+      }
+    },
+    {
+      propertyName: 'createdBy',
+      title: 'Created by',
+      className: 'te-modal__table-cell'
+    },
+    // Todo: Fix back end to send
+    // dateCreated data
+    // {
+    //   propertyName: 'dateCreated',
+    //   title: 'Date Created'
+    // },
+    {
+      template: 'custom/tableDelete',
+      title: '',
+      className: 'te-modal__table-cell te-modal__table-cell--delete te-modal__table-cell--dark'
+    }
+  ],
+
+  /**
+   * Custom classes to be applied to the entity modal table
+   */
+  classes: {
```

Comment:
@ttbach This is how you override some of the table css. this is passed inside the `ember-models-table` in https://github.com/linkedin/pinot/pull/2575/files#diff-fa2088e19d4061108d3a57d8e9aeefbaR77